### PR TITLE
Another set of features

### PR DESCRIPTION
**Pull request by Malacath-92**
This time I updated the Readme.md to specify the new features. Of particular importance are the following for me:

* I extensively tested the *useInShaderTextures* option and don't think it should be considered experimental at this point, though some user input would be nice. I don't know how many users the extension has right now though,
* I added the ability to load other shaders as textures, which required a significant refactor to be at all readable and usable. These can be specified inside shaders or settings, though in my opinion the latter version should just as well be deprecated because it is not nearly as user-friendly at this point. Shaders feeding back into themselves is not supported yet,
* paths can now be specified relative to the working directory, this is handled by the extension itself and doesn't rely on third-party behavior (hopefully),
* last but not least improved the interface for compilation errors, which now highlights the line an error happens on, when the user clicks said line. This works as expected with multiple passes, in that it displays all errors on one page and highlights lines in the appropriate file.

One particular problem that I would like to address, but have not been able to, is the problematic sampling of textures. If you investigate the _uv-warp.glsl -> multipass.glsl_ demo and output only _uv_ in _uv-warp.glsl_ then you can see that using the resulting texture will have significant artifacts compared to computing _uv_ yourself inside _multipass.glsl_. I tried eliminating it by using different filtering methods but to no avail. Maybe someone else has an idea?

### CODE REVIEW

**Comment by Malacath-92** [comment]
I also now just updated _three.js_ and passed a webgl2 context to WebGlRenderer. If not available this will automatically fall back to webgl1, but when available it will allow more modern shaders.

File: src/preview.html [hunk 1]
```diff
@@ -1,142 +1,201 @@
...
-        uniform sampler2D   iChannel3;
-        uniform sampler2D   iChannel4;
-        uniform sampler2D   iChannel5;
-        uniform sampler2D   iChannel6;
-        uniform sampler2D   iChannel7;
-        uniform sampler2D   iChannel8;
-        uniform sampler2D   iChannel9;      
+</body>
+<script src="file://d:\zollk\Documents\Visual Studio 2017\Projects\shader-toy\resources\jquery.min.js"></script>
+<script src="file://d:\zollk\Documents\Visual Studio 2017\Projects\shader-toy\resources\three.min.js"></script>
```
Comment (stevensona) [inline src/preview.html]: Absolute paths aren't going to work here.

**Comment by stevensona** [comment]
Thanks @Malacath-92. This looks great. Really powerful features.

File: src/preview.html (hunk 1 above)
Comment (Malacath-92) [reply src/preview.html]: Yeah, true, my bad. I'm gonna have to fix that when I'm home. Not sure what the original said there, probably had a url to the min.js versions of those.

Comment (Malacath-92) [reply src/preview.html]: Instead I just copied the result of the whole parsing process.

Comment (Malacath-92) [reply src/preview.html]: Fixed this and also added local stats.js because I was working mainly without internet and couldn't check performance  :D

**Comment by Malacath-92** [comment]
Added three more commits

**Comment by stevensona** [comment]
Great, thank you. I will merge/release tonight.

**Comment by Malacath-92** [comment]
Any idea though what the sampling issue is related to?

**Comment by stevensona** [comment]
No clue. Haven't looked into it yet. Would you mind opening up a new issue with a screenshot of the problem?

**Comment by Malacath92** [comment]
I was gonna do that on the main repo as soon as this gets merged. Otherwise I'll keep investigating before you merge if you would prefer that.

**Comment by stevensona** [comment]
I was going to go ahead and merge it with the understanding this would be an outstanding issue with the new features. The new features are important enough I don't want to block the release. What do you think?

**Comment by stevensona** [comment]
Released 0.7.1. Thanks @Malacath-92

**Comment by Malacath92** [comment]
Yes, I approve that the issue isn't so severe that it should block the release. I'll open the Issue later when I get to a PC.

**Pull request merged**